### PR TITLE
Def parser fix for definitions with no header

### PR DESCRIPTION
**Pull request by ikaneshiro**
Def parser scanner returns the first section it runs into if there isn't a header, which was causing an index out of range panic. This detects if the first returned token is a header or not and parses it accordingly.

### CODE REVIEW

File: src/pkg/build/build.go [hunk 1]
```diff
@@ -243,23 +291,23 @@ func (b *Build) runBuildEngine() error {
 
 // Bundle creates the bundle using the ConveyorPacker and returns it. If this
 // function is called multiple times it will return the already created Bundle
-func (b *Build) Bundle() (*types.Bundle, error) {
-	if b.b != nil {
-		return b.b, nil
-	}
+// func (b *Build) Bundle() (*types.Bundle, error) {
```
Comment (jchavezlavalle) [inline src/pkg/build/build.go]: There's commented code in here @ikaneshiro

File: src/pkg/build/build.go [hunk 2]
```diff
@@ -243,23 +291,23 @@ func (b *Build) runBuildEngine() error {
...
 
-	if err := b.c.Get(b.d); err != nil {
-		return nil, fmt.Errorf("conveyor failed to get: %v", err)
-	}
+// 	if err := b.c.Get(b.b); err != nil {
+// 		return nil, fmt.Errorf("conveyor failed to get: %v", err)
+// 	}
```
Comment (jchavezlavalle) [inline src/pkg/build/build.go]: and here also more commented code

File: src/pkg/build/build.go [hunk 3]
```diff
@@ -243,23 +291,23 @@ func (b *Build) runBuildEngine() error {
...
 
-	bundle, err := b.c.Pack()
-	if err != nil {
-		return nil, fmt.Errorf("packer failed to pack: %v", err)
-	}
+// 	bundle, err := b.c.Pack()
+// 	if err != nil {
+// 		return nil, fmt.Errorf("packer failed to pack: %v", err)
+// 	}
```
Comment (jchavezlavalle) [inline src/pkg/build/build.go]: also here some commented code

File: src/pkg/build/build.go [hunk 4]
```diff
@@ -243,23 +291,23 @@ func (b *Build) runBuildEngine() error {
...
 
-	b.b = bundle
-	return b.b, nil
-}
+// 	b.b = bundle
+
+// 	b.addOptions()
+
+// 	return b.b, nil
+// }
```
Comment (jchavezlavalle) [inline src/pkg/build/build.go]: as well as this part w/ some commented code

**Review by jchavezlavalle — COMMENTED** [review]
just added some comments, there's some commented code, guess it should be removed

File: src/pkg/build/types/definition_parser_deffile_test.go [hunk 5]
```diff
@@ -135,7 +136,8 @@ func TestParseDefinitionFileFailure(t *testing.T) {
 			}
 			defer defFile.Close()
 
-			if _, err = ParseDefinitionFile(defFile); err == nil {
+			if d, err := ParseDefinitionFile(defFile); err == nil {
+				fmt.Println(d)
```
Comment (bauerm97) [inline src/pkg/build/types/definition_parser_deffile_test.go]: Purpose of `fmt.Println()` here? Forgot to remove a debug statement? We should rather be using a `t.Log` statements to log here

**Review by bauerm97 — CHANGES_REQUESTED** [review]

File: src/pkg/build/types/definition_parser_deffile_test.go (hunk 5 above)
Comment (ikaneshiro) [reply src/pkg/build/types/definition_parser_deffile_test.go]: missed debug statement

**Review by bauerm97 — APPROVED** [review]

**Pull request merged**